### DISCUSSION TITLE
How to get manifest file

### DISCUSSION THREAD

**Discussion by imranmehdi5511** [gramineproject/gramine#1365] [General]
I have a aes256 encryption code written in c programming language, How do I get the makefile and manifest file for it? How to create these files or automatically create them? I have been searching for months but no detailed answer on this topic

**Comment by kailun-qin** [comment]
Hi @imranmehdi5511!

> How do I get the makefile and manifest file for it? How to create these files or automatically create them?

For Makefiles, there's nothing special in working w/ Gramine, so you have to either create it on your own or use some build systems like CMake. Basically, you could just leverage your original Makefiles of this `aes256 encryption code written in c programming language` and sign the manifest with `gramine-sgx-sign` tool, see https://gramine.readthedocs.io/en/latest/manpages/gramine-sgx-sign.html.  

For manifests, unfortunately there is no such auto-generation tool available in Gramine, i.e., you have to create your application-specific manifest file on your own by following https://gramine.readthedocs.io/en/latest/manifest-syntax.html. There are acutally a bunch of application examples available here: https://github.com/gramineproject/gramine/tree/master/CI-Examples. So you could start with a thoroughly documented example of Redis (https://github.com/gramineproject/gramine/blob/master/CI-Examples/redis/redis-server.manifest.template) to understand manifest options.
